extensions/resteasy-classic/resteasy-client/runtime/src/main/java/io/quarkus/restclient/runtime/QuarkusProxyInvocationHandler.java

```diff
@@ -299,4 +382,20 @@ private static Annotation[] merge(List<Annotation> methodLevelBindings, List<Ann
         return merged.toArray(new Annotation[] {});
     }
 
+    private static ClassLoader getClassLoader(final Class<?> type) {
+        if (System.getSecurityManager() == null) {
```

Comment:
I don't think we need to handle SM or priviledged actions in Quarkus; you should be able to safely remove that.

Reply:
so just `type.getClassLoader()` ? as this only has just one usage I would directly inline it

Reply:
in createProxy:273 
`final Object proxy = Proxy.newProxyInstance(resourceInterface.getClassLoader(), interfaces,
                new ProxyInvocationHandler(resourceInterface, target, Set.copyOf(providers), client));`